src/main/java/seedu/address/model/scheduler/time/util/TimeUtil.java

```diff
@@ -89,4 +93,39 @@ public static String formatLocalTime(LocalTime time) {
             return String.format("%d PM", hourOfDay % 12);
         }
     }
+
+    /**
+     * Checks that amongst these timeperiods, if there are any clashes based on timing
+     * and not activity.
+     */
+    public static boolean hasAnyClash(List<TimePeriod> timePeriods) {
+        List<Pair<Integer, TimePeriod>> indexedTimePeriods = MathUtil.<TimePeriod>indexObjects(timePeriods);
+        List<Pair<Integer, TimePeriod>> second = List.copyOf(indexedTimePeriods);
+        List<List<Pair<Integer, TimePeriod>>> cartesianProductPairs =
+            MathUtil.<Pair<Integer, TimePeriod>>getCartesianProduct(indexedTimePeriods, second);
+        Stream<List<Pair<Integer, TimePeriod>>> cartesianProductTimePeriodStream = cartesianProductPairs.stream()
+            .filter(listPair -> !listPair.get(0).getKey().equals(listPair.get(1).getKey()))
+            .filter(listPair -> listPair.get(0).getValue().hasClash(listPair.get(1).getValue()));
```

Comment:
I think for these multi-layered stream uses, it would be better to leave a space in between, so that it is more readable. Stream types get crazy sometimes. Also, might want to extract out some methods (though I'm not sure how).

Reply:
This design only came about as there will be an "arrow" visible due to the nesting. But it is part of my todos